src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -319,6 +320,14 @@ def output_dict(self) -> Dict[str, Any]:
             }
         return self._output_dict
 
+    @cached_property
+    def plan_height(self) -> int:
+        return 1 + max((child.plan_height for child in self.children), default=0)
+
+    @cached_property
+    def num_duplicate_nodes(self) -> int:
+        return len(find_duplicate_subtrees(self))
```

Comment:
`find_duplicate_subtrees` looks expensive. This seems to run every time a `SnowflakePlan` is created?

Reply:
this is called when evaluating `df_collect_api_telemetry` so only when action statement is called.

Reply:
We can do a performance tests for the client side.